File: Leetcode Questions/Daily Question/05-04-1544-Make-the-String-Great/1544.py
def makeGoodStr(string: str) -> str:
    stack = []
    for char in string:
        if stack and abs(ord(char) - ord(stack[-1])) == 32:
            stack.pop()
        else:
            stack.append(char)
    return ''.join(stack)

print(makeGoodStr('sSstAaackKCck'))

File: Leetcode Questions/Daily Question/05-04-1544-Make-the-String-Great/1544.cpp
#include<iostream>
#include<stack> 
using namespace std;


string makeGoodStr(string s) {
    stack<char> stack;
    for(char c: s) {
        if(!stack.empty() && abs(c - stack.top()) == 32) {
            stack.pop();
        }
        else {
            stack.push(c);
        }
    }
    string result;
    while (!stack.empty()) {
        result = stack.top() + result;
        stack.pop();
    }
    return result;
}

int main() {
    cout<<makeGoodStr("sSstAaackKCck");
    return 0;
}